src/stk/math/types.h

```diff
@@ -126,7 +128,38 @@ struct STK_ALIGN(16) double4
 };
 
 
-#endif
+#endif // STK_USE_CUDA
+
+
+template<typename T, unsigned int rows_, unsigned int cols_>
+struct matrix
+{
+    T _data[rows_][cols_];
```

Comment:
This could also be changed to a 1D array. Not sure how much of a difference it will make though, since index arithmetic will be heavily optimised by the compiler in any case. I guess the optimal solution depends on which scenarios we will use the matrix.

Reply:
One thing I typically did for matrices (fixed size) was to define it as n vectors. That way you could easily perform vector operators on the data and made it easy to SIMD. I don't think this is an issue in modern compilers though so this should work fine.